source/winUser.py

```diff
@@ -514,6 +514,10 @@ def FindWindow(className, windowName):
 IDCANCEL=3
 
 def MessageBox(hwnd, text, caption, type):
+	if hasattr(text, 'decode'):
+		text = text.decode('mbcs')
+	if caption and hasattr(caption, 'decode'):

```

Comment:
You can safely use hasattr on a None, so the if caption check is not required. Furthermore, also avoid using hasattr here and use an instance check instead.

Reply:
Actually, at least on python 3, I'm pretty sure that you *must* ommit the `if caption` check. If you don't and you provide an empty byte string, ctypes might throw an error, telling you that you can't throw an empty byte string into a function that expects a wide character string.

Reply:
nvda_snapshot_pr8771-16091,73deebfe.exe is tested as well.